osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -148,15 +153,47 @@ public void CopyFrom(Mod source)
             if (source.GetType() != GetType())
                 throw new ArgumentException($"Expected mod of type {GetType()}, got {source.GetType()}.", nameof(source));
 
-            foreach (var (_, prop) in this.GetSettingsSourceProperties())
+            foreach (var (_, property) in this.GetSettingsSourceProperties())
             {
-                var targetBindable = (IBindable)prop.GetValue(this)!;
-                var sourceBindable = (IBindable)prop.GetValue(source)!;
+                var targetBindable = (IBindable)property.GetValue(this)!;
+                var sourceBindable = (IBindable)property.GetValue(source)!;
 
                 CopyAdjustedSetting(targetBindable, sourceBindable);
             }
         }
 
+        /// <summary>
+        /// When converting mods from one ruleset to the other, this method makes sure
+        /// to also copy the values of all settings sharing same <see cref="MemberInfo.Name"/> between the two instances.
+        /// </summary>
+        /// <remarks>Copied values are unchanged, even if they have different clamping ranges.</remarks>
+        /// <param name="source">The mod to extract settings from.</param>
+        public void CopyCommonSettings(Mod source)
+        {
+            if (source.UsesDefaultConfiguration)
+                return;
+
+            foreach (var (name, targetSetting) in Settings)
+            {
+                if (!source.Settings.TryGetValue(name, out IBindable? sourceSetting))
+                    continue;
+
+                if (sourceSetting.IsDefault)
+                    continue;
+
+                var targetType = targetSetting.GetType();
+                var sourceType = sourceSetting.GetType();
+
+                if (!targetType.IsAssignableFrom(sourceType) && !sourceType.IsAssignableFrom(targetType))
+                    continue;
```

Comment:
Why is this checking the assignability in *both* directions? Isn't `targetType.IsAssignableFrom(sourceType)` the only part that matters? Also, why `&&`? Is this intending to say `targetType == sourceType`? Really confused.

Reply:
I agree the inverted logic is a bit confusing. Basically if either of the `IsAssignableFrom` checks is true, then the settings get changed. If one of them is `BindableInt` and the other is `Bindable<int>` then one of the checks will be true.